File: Scala/day18.scala
import scala.collection.mutable.Stack
import scala.collection.mutable.Queue

class Palindrome() {
  // Write your code here
  var stack = new Stack[Char]()
  var queue = new Queue[Char]()
  def pushCharacter(ch: Char): Unit = {
    stack.push(ch)
  }
  def enqueueCharacter(ch: Char): Unit = {
    queue.enqueue(ch)
  }
  def popCharacter(): Char = {
    return stack.pop()
  }
  def dequeueCharacter(): Char = {
    return queue.dequeue()
  }
}

object Solution {
  def main(args: Array[String]) {
    // read the string s
    var s = scala.io.StdIn.readLine()
    // create the Palindrome class object p
    var p = new Palindrome()
    var i = 0
    var len = s.length()
    // push all the characters of string s to stack
    for (i <- 0 to len-1) {
      p.pushCharacter(s.charAt(i))
    }
    // enqueue all the characters of string s to queue
    for (i <- 0 to len-1) {
      p.enqueueCharacter(s.charAt(i))
    }
    var f = true
    /*pop the top character from stack
      dequeue the first character from queue
      compare both the characters*/
    for (i <- 0 to len-1) {
      if (p.popCharacter() != p.dequeueCharacter()) f = false
    }
    // finally print whether string s is palindrome or not
    if (f) println("The word, " + s + ", is a palindrome.")
    else println("The word, " + s + ", is not a palindrome.")
  }
}

File: Java/day18.java
import java.io.*;
import java.util.*;

class Palindrome
{
    // Write your code here
    private Deque<Character> stack;
    private Queue<Character> queue;

    public Palindrome() {
        stack = new LinkedList<>();
        queue = new LinkedList<>();
    }

    public void pushCharacter(char ch) {
        this.stack.push(Character.valueOf(ch));
    }

    public void enqueueCharacter(char ch) {
        this.queue.add(Character.valueOf(ch));
    }

    public char popCharacter() {
        return this.stack.pop().charValue();
    }

    public char dequeueCharacter() {
        return this.queue.poll().charValue();
    }
}

public class day18
{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();
        
        Palindrome p = new Palindrome();
        char arr[] = s.toCharArray();
        for (char c : arr) {
            p.pushCharacter(c);
        }
        
        for (char c : arr) {
            p.enqueueCharacter(c);
        }
        
        boolean f = true;
        
        for (int i = 0; i < s.length(); i++) {
            if (p.popCharacter() != p.dequeueCharacter()) {
                f = false;                
                break;
            }
        }
        
        if (f) {
            System.out.println("The word, " + s + ", is a palindrome.");
        } else {
            System.out.println("The word, " + s + ", is not a palindrome.");
        }
    }
}